airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/RedshiftStagingS3Destination.java

```diff
@@ -130,7 +130,7 @@ public AirbyteMessageConsumer getConsumer(final JsonNode config,
         getDatabase(getDataSource(config)),
         new RedshiftS3StagingSqlOperations(getNamingResolver(), s3Config.getS3Client(), s3Config, encryptionConfig),
         getNamingResolver(),
-        CsvSerializedBuffer.createFunction(null, () -> new FileBuffer(CsvSerializedBuffer.CSV_GZ_SUFFIX)),
+        CsvSerializedBuffer.createFunction(null, () -> new FileBuffer(CsvSerializedBuffer.CSV_GZ_SUFFIX, catalog.getStreams().size())),
```

Comment:
```suggestion
        CsvSerializedBuffer.createFunction(null, () -> new FileBuffer(CsvSerializedBuffer.CSV_GZ_SUFFIX)),
```
I wouldn't set it to the size of the catalog because it makes the limit useless and potentially dangerous for running out of file descriptors.

There is a more global discussion to be had, it's not just specific to Redshift: ie you could change the [constant value](https://github.com/airbytehq/airbyte/blob/b261d1fe77ab4cb4c99253cdef0e884c017507aa/airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/record_buffer/FileBuffer.java#L35) instead. 

But I agree that the limit could be set much higher than just 10, but it shouldn't be more than 10000 or something more reasonable

Reply:
I agree this is broader than just Redshift - my original PR addressed all destinations that used buffering, and then I was asked to split it up. I think we lost some context that way.

@ChristopheDuong is there a maximum number of streams supported by a CDC source? Or does bumping it up to a much larger constant just reduce the probability of encountering this issue?

Perhaps we increase the constant, but then still pass the stream count so we can log a warning about reduced performance if it exceeds? We could take the min of the value passed to the constructor or the constant.

Reply:
@adam-bloom I'd be happy if we set the constant to 100 or even make it configurable so the users can set their own value

Reply:
@alexnikitchuk configurable is interesting - I hadn’t considered that! I think that’s important from my comment above is helping users identify the “correct” configuration. Some sort of log that they are misconfigured may be very helpful, since currently you’d need to recognize the constant small flushes - there isn’t a clear log indicating the stream count mismatch.

Reply:
@adam-bloom yup, a warning message in the logs about performance implications would be great to have